app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.kt

```diff
@@ -0,0 +1,169 @@
+package fr.free.nrw.commons.utils
+
+import android.app.Activity
+import android.app.AlertDialog
+import android.app.Dialog
+import android.content.DialogInterface
+import android.view.View
+import fr.free.nrw.commons.R
+import timber.log.Timber
+
+object DialogUtil {
+    /**
+     * Shows a dialog safely.
+     * @param activity the activity
+     * @param dialog the dialog to be shown
+     */
+    private fun showSafely(activity: Activity?, dialog: Dialog?) {
+
+        if (activity == null || dialog == null) {
+            Timber.d("Show called with null activity / dialog. Ignoring.")
+            return
+        }
+
+        if (activity.isFinishing || activity.isDestroyed) {
+            Timber.e("Activity is not running. Could not show dialog. ")
+            return
+        }
+        try {
+            dialog.show()
+        } catch (e: IllegalStateException) {
+            Timber.e(e, "Could not show dialog.")
+        }
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        onPositiveBtnClick: Runnable?,
+        onNegativeBtnClick: Runnable?
+    ) {
+        createAndShowDialogSafely(
+            activity,
+            title,
+            message,
+            activity.getString(R.string.yes),
+            activity.getString(R.string.no),
+            onPositiveBtnClick,
+            onNegativeBtnClick
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String,
+        negativeButtonText: String,
+        onPositiveBtnClick: Runnable?,
+        onNegativeBtnClick: Runnable?
+    ) {
+        createAndShowDialogSafely(
+            activity,
+            title,
+            message,
+            positiveButtonText,
+            negativeButtonText,
+            onPositiveBtnClick,
+            onNegativeBtnClick
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        onPositiveBtnClick: Runnable?,
+        onNegativeBtnClick: Runnable?,
+        customView: View,
+        cancelable: Boolean
+    ) {
+        createAndShowDialogSafely(
+            activity,
+            title,
+            message,
+            activity.getString(R.string.yes),
+            activity.getString(R.string.no),
+            onPositiveBtnClick,
+            onNegativeBtnClick,
+            customView,
+            cancelable
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String?,
+        onPositiveBtnClick: Runnable?,
+        cancellable: Boolean
+    ) {
+        createAndShowDialogSafely(
```

Comment:
with named [arguments](https://kotlinlang.org/docs/reference/functions.html#named-arguments) you can omit specifying an argument if it is the same as the default value

Reply:
Thanks, Changed made!